Update docs for delete data stream API to show that multiple names are supported (#58838)
elastic/elasticsearch: docs/reference/indices/delete-data-stream.asciidoc

`<data-stream>`::
(Required, string)

> Comma-separated list of names or wildcard expressions of data streams to
> delete.

Reviewer: I've been inconsistent with this, but `or` is technically incorrect as a list of wildcard expressions is supported. I'll open a PR to fix this in the other data stream API docs.
```suggestion
Comma-separated list of data streams to delete.
Wildcard (`*`) expressions are supported.
```
